### ISSUE TITLE
`casts()`-method array form still resolves to `mixed` (CastsMethodParser has no Array_ branch)

### ISSUE THREAD

**Issue by alies-dev** [psalm/laravel-psalm-plugin#1289]
### Summary

`ensureCastsAreStringValues()` normalizes **both** halves of `array_merge($this->casts, $this->casts())`. #1281 closed the declared-`$casts` half by calling the framework method at warm-up. The `casts()` half is unaffected: warm-up never executes `casts()`, so its values reach the registry through `CastsMethodParser`, which cannot resolve an array literal.

`resolveStringValue()` handles only `String_`, `::class`, and `Concat`. An `Array_` node returns `null`, and `extractCastsFromArray()` maps that to `'mixed'` (`$casts[$key] = $value ?? 'mixed'`).

```php
protected $casts = ['options' => [AsCollection::class, Collection::class]];        // resolves (#1281)
protected function casts(): array {
    return ['options' => [AsCollection::class, Collection::class]];               // -> 'mixed'
}
protected function casts(): array {
    return ['options' => AsCollection::of(Collection::class)];                    // -> 'mixed' (StaticCall)
}
```

Runtime holds `AsCollection:Illuminate\Support\Collection` in the first two. `casts()` is the form Laravel's docs have recommended since 11, and the only one that can express `AsCollection::of(...)` at all.

### Severity

Precision only, not correctness — `'mixed'` is a sound widening, so no false positives and no section drop. This is strictly the quiet twin of #1281's crash.

### Fix sketch

Add an `Array_` branch to `resolveStringValue()` mirroring the framework's `is_array` arm: single element collapses to itself, otherwise first element is the base and the rest join with `,` after a `:` (keys dropped — `implode()` takes values). Return `null` (→ `'mixed'`) for anything not reproducible: an unresolvable element, a spread, `[]`, or a single element with an explicit key (Laravel returns `$cast[0]`, which such an array lacks).

Unlike #1281 this must COPY the framework rather than call it — there is no instance, so there is no property to normalize. Affordable only because the `is_array` arm is byte-identical across `illuminate/database: ^12.14 || ^13.3` (verified at 12.14.0 / 12.25.0 / 12.26.0 / 13.3.0 / 13.20.0).

### Testing note (the reason this was split out)

**No suite currently resolves a cast to a property type.** The type suite ships no migrations, so every cast there is inert (`App\Models\AbstractDocument`'s own docblock says so); unit registry tests skip the AST merge because `codebaseMethodsInitialized()` is false under `newInstanceWithoutConstructor()`; `laravel-test.sh` wires no migrations for these models either. A prototype with a `casts()` array-form model resolved to `mixed` for BOTH the array form and a plain-string control — the harness, not the parser.

So this needs either a schema seam in the type suite, or a direct test of the collapse against a real AST. Worth deciding before implementing.

### Prevalence

Zero array-form casts across the 17 public benchmark apps (vs 2,344 string-form). Low urgency.

**Comment by alies-dev** [comment]
Closing as **won't fix**. Three reasons, in order of weight:

**1. Zero prevalence.** Measured across the 17 public benchmark apps (aureuserp, bagisto, coolify, monica, pixelfed, filament, unit3d, vito, …): **0** array-form cast entries in either the `$casts` property or the `casts()` method, against 2,344 string-form. The fluent helpers Laravel actually documents (`AsCollection::of()`, `AsEnumCollection::of()`) all return strings, which is why the array form is effectively unused in the wild.

**2. It degrades safely.** `'mixed'` is a sound widening: no crash, no section drop, no false positive. Just lost precision on one key. That is categorically different from #1281, which silently withheld the model's *entire* casts section.

**3. No test seam exists, and building one costs more than the fix.** No suite in this repo resolves a cast to a property type: the type suite ships no migrations, so every cast there is inert (see `App\Models\AbstractDocument`'s docblock); unit registry tests skip the AST merge because `codebaseMethodsInitialized()` is false under `newInstanceWithoutConstructor()`; `laravel-test.sh` wires no migrations for these models either. A prototype confirmed this empirically: a `casts()` array-form model resolved to `mixed` for **both** the array form and a plain-string control — the harness, not the parser.

So the fix would be ~15 lines guarded by either new test infrastructure or a reflection test against a private helper, to close a precision gap no benchmarked app can reach.

Reopen if a real project turns up using the array form in `casts()` — the fix sketch above still stands, and the `is_array` arm it mirrors is byte-identical across `illuminate/database: ^12.14 || ^13.3`, so it would need no version gate.

Related: #1290 (the same normalizer's `default` arm, which *does* have a reachable severe failure mode and is worth doing).
